### PR TITLE
Calibrate cjc at next boot

### PR DESCRIPTION
**Pull request by discip**
After reading this https://github.com/Ralim/IronOS/issues/1387 post, I had the idea to make CJC (Cold Junction Compensation) calibration as accurate as possible by executing it at boot.

This is my try on solving that one.

Tested this on my **`TS80P`**. 
However, on the **`Pinecil`** it does ***not work!*** 🤷‍♂️

1. What has changed:
   1. **`Calibrate temperature`** is now called **`Calibrate CJC at next boot`**.
   2. It is now performed as the name implies at the next boot.
   3. This setting is a single shot only (You have to reenable it after every reboot.).
   4. There is one more entry in the **`Debug Menu`** named **`Tip Off`**.
   5. All (at least I hope so 😅) the corresponding documentations have been updated.

2. What is currently not working (at least for the):
   1. I tried to implement the condition that the **tip has to be connected** and
   2. the condition that the **tip temperature has to be less than 31°C** in order to make the calibration happen.
https://github.com/discip/IronOS/blob/7fd2b4e9cbe6ac490b870e494c81f78823ac81cc/source/Core/Threads/GUIThread.cpp#L1005
But for some reason this does not work.

3. What I don't really know how to achieve efficiently:
* Make the previous warning message appear before the checkbox gets checked.

### CODE REVIEW

File: source/Core/Threads/GUIThread.cpp [hunk 1]
```diff
@@ -996,6 +1000,40 @@ void startGUITask(void const *argument) {
   }
 #endif
 #endif
+  // al_qu Calibrate Cold Junction Compensation directly at boot, before internal components get warm.
+  uint32_t Temp = TipThermoModel::getTipInC();
+  if (!isTipDisconnected() && Temp <= 30 && getSettingValue(SettingsOptions::CalibrateCJC) > 0) {
```
Comment (discip) [inline source/Core/Threads/GUIThread.cpp:1005]: @Ralim
First of all:
Do you think this approach makes sense?

And if so, do you have time to look into this?
Especially this line here.

Comment (discip) [inline source/Core/Threads/GUIThread.cpp:1005]: @Ralim
First of all:
Do you think this approach makes sense?

And if so, do you have time to look into this?
Especially this line here:
```C#
if (!isTipDisconnected() && Temp <= 30 && getSettingValue(SettingsOptions::CalibrateCJC) > 0)
```

File: source/Core/Threads/GUIThread.cpp [hunk 2]
```diff
@@ -996,6 +1000,40 @@ void startGUITask(void const *argument) {
   }
 #endif
 #endif
+  // Calibrate Cold Junction Compensation directly at boot, before internal components get warm.
+  uint32_t Temp = TipThermoModel::getTipInC();
+  if (!isTipDisconnected() && Temp <= 30 && getSettingValue(SettingsOptions::CalibrateCJC) > 0) {
```
Comment (discip) [inline source/Core/Threads/GUIThread.cpp:1005]: @Ralim
First of all:
Do you think this approach makes sense?

And if so, do you have time to look into this?
Especially this line:
```C#
if (!isTipDisconnected() && Temp <= 30 && getSettingValue(SettingsOptions::CalibrateCJC) > 0)
```

Comment (Ralim) [reply source/Core/Threads/GUIThread.cpp:1005]: Hmm okay, this makes sense.
Though you will need to have a 
```
while (preStartChecks() == 0) {
    ulTaskNotifyTake(pdTRUE, TICKS_100MS);
  }
```

Block in here to block the code progressing past this before the ADC is stable, and other tip calibration stuff is done. (See https://github.com/Ralim/IronOS/blob/434ca2647ace5bf68a4a543775cce4e9807f7f10/source/Core/Threads/PIDThread.cpp#L49 for where I copied that from)

I would yank this block of logic you have here out into a function like `performTipOffsetCalibration()`

And then just call that if the function is set.

Mostly for cleaner code paths etc.

**Comment by Ralim** [comment]
>What I don't really know how to achieve efficiently:
>Make the previous warning message appear before the checkbox gets checked.

efficiently is probably the magic word here.
You could keep the custom on-set handler, and instead prompt warning; if they accept _then_ set the setting to 1.
That should get it done the easiest code-wise

**Pull request closed without merging**